amads/harmony/root_finding/parncutt.py

```diff
@@ -0,0 +1,190 @@
+        "v2": {0: 10, 7: 5, 4: 3, 10: 2, 2: 1},
+    }
+
+    def __init__(
+        self,
+        chord: List[int],
+        root_support_weights: Union[str, Dict[int, float]] = "v2",
+        exponent: float = 0.5,
+    ):
+        if len(chord) == 0:
+            raise ValueError("Chord must contain at least one pitch")
+
+        self.chord = chord
+        self.pc_set = set([pitch % 12 for pitch in chord])
+        self.root_support_weights = self.load_root_support_weights(root_support_weights)
+        self.exponent = exponent
+        self.root_strengths = [self.get_root_strength(pc) for pc in range(12)]
+        self.root = self.get_root()
+        self.root_ambiguity = self.get_root_ambiguity()
+
+    def load_root_support_weights(
+        self, root_support_weights: Union[str, Dict[int, float]]
+    ) -> Dict[int, float]:
+        if isinstance(root_support_weights, str):
+            if root_support_weights not in self.available_root_support_weights:
+                raise ValueError(
+                    f"Unknown root support weights version: {root_support_weights}"
+                )
+        return self.available_root_support_weights[root_support_weights]
+
+    def get_root_strength(self, pc: int) -> float:
+        return sum(
+            support_weight
+            for interval, support_weight in self.root_support_weights.items()
+            if (pc + interval) % 12 in self.pc_set
+        )
+
+    def get_root(self) -> int:
+        return self.root_strengths.index(max(self.root_strengths))
+
+    def get_root_ambiguity(self) -> float:
+        max_weight = max(self.root_strengths)
+        if max_weight == 0:
+            return 0.0
+        return sum(w / max_weight for w in self.root_strengths) ** self.exponent
+
+    def visualize(self, title: Optional[str] = None) -> None:
+        """
+        Visualize the root support weights for a chord.
+
+        Parameters
+        ----------
+        title : Optional[str], optional
+            Title for the plot, by default None.
+
+        Examples
+        --------
+        >>> analysis = ParncuttRootAnalysis([0, 4, 7])
+        >>> analysis.visualize()
+        """
+        # Create a bar chart
+        plt.figure(figsize=(10, 6))
+        bars = plt.bar(range(12), self.root_strengths)
+
+        # Highlight the root
+        bars[self.root].set_color("red")
+
+        # Add labels
+        plt.xlabel("Pitch class")
+        plt.ylabel("Root strength")
+
+        # Add pitch class names
+        pitch_class_names = [
+            "C",
+            "C#",
+            "D",
+            "D#",
+            "E",
+            "F",
+            "F#",
+            "G",
+            "G#",
+            "A",
+            "A#",
+            "B",
+        ]
+        plt.xticks(range(12), pitch_class_names)
+
+        # Add title
+        if title is None:
+            chord_str = ", ".join(
+                str(pc) for pc in sorted(set(p % 12 for p in self.chord))
```

Comment:
Set does sorting. 

```py
>>> a = [3, 1, 2]
>>> set(a)
{1, 2, 3}
```

Reply:
I don't think this is true, actually, see e.g. https://stackoverflow.com/questions/3812429/is-pythons-set-stable

I think it's safest to explicitly sort here!

Reply:
Huh! Interesting. Thanks.